comid/tdx-profile/teetcbstatus.go

```diff
@@ -0,0 +1,51 @@
+// nolint:dupl
+package tdx
+
+import "fmt"
+
+type TeeTcbStatus setType
+
+func NewTeeTcbStatus(val []any) *TeeTcbStatus {
+	var ts TeeTcbStatus
+	if len(val) == 0 {
+		return nil
+	}
+
+	for _, v := range val {
+		switch t := v.(type) {
+		case string:
+			ts = append(ts, t)
+		default:
+			return nil
+		}
+	}
+	return &ts
+}
+
+func (o *TeeTcbStatus) AddTeeTcbStatus(val []any) error {
+	for _, v := range val {
+		switch t := v.(type) {
+		case string:
+			*o = append(*o, t)
+		default:
+			return fmt.Errorf("invalid type: %T for tcb status", t)
+		}
+	}
+	return nil
+}
+
+func (o TeeTcbStatus) Valid() error {
```

Comment:
Why change your receiver type from *TeeTcbStatus to TeeTcbStatus?

Reply:
The philosophy is that any Method that does not modify the stuff uses Non pointer receiver!